Makefile

```diff
@@ -4,33 +4,45 @@ SETUP = $(PYTHON) setup.py
 TESTRUNNER ?= unittest
 RUNTEST = PYTHONHASHSEED=random PYTHONPATH=$(shell pwd)$(if $(PYTHONPATH),:$(PYTHONPATH),) $(PYTHON) -m $(TESTRUNNER) $(TEST_OPTIONS)
 COVERAGE = python3-coverage
+EDITABLE := $(shell pip show dulwich 2>/dev/null | grep -E "^Editable project location:")
```

Comment:
This doesn't work, it will suggest you're in an editable installation if there's a system-wide install of dulwich. 

pip maybe also not be available as "pip" but only as "pip3"

I'd keep this simple and avoid suggest running the Makefile to invoke the test suite in CONTRIBUTING.rst, but just running "python -m unittest dulwich.tests.test_suite".

Reply:
On Debian systems where there's packages that use Dulwich this is quite common. This is the case on my own system, and running "pip show dulwich" here picks up the system version of Dulwich. This is breaking my personal workflow with the Makefile, because now it stops building the extensions.

I'd be inclined to suggest that CONTRIBUTING.rst doesn't reference the Makefile at all and just suggests running plain commands. That matches what's in the CI workflow, and addresses your concerns about the Make targets. And then CI can avoid touching the Makefile.